src/firebase-utils.ts

```diff
@@ -139,6 +158,17 @@ export function initializeFirebase(
         process.env.FIREBASE_AUTH_EMULATOR_HOST,
       );
       /* eslint-enable no-console */
+    } else if (protectProduction?.auth && protectProduction.auth !== 'none') {
```

Comment:
If we can drop optional chaining I believe that will preserve support for older cypress versions
```suggestion
    } else if (protectProduction && protectProduction.auth && protectProduction.auth !== 'none') {
```